src/xgov-dapp/src/shared/hooks/useOverflow.ts

```diff
@@ -1,25 +1,19 @@
-import React, { useLayoutEffect } from 'react'
-import type { MutableRefObject } from 'react'
-
-export const useOverflow = (ref: MutableRefObject<HTMLElement | null>, callback?: (hasOverflow: boolean) => boolean) => {
-  const [isOverflow, setIsOverflow] = React.useState<boolean | undefined>(undefined)
+import { useLayoutEffect, useRef, useState } from 'react'
 
+export const useOverflow = (onChange?: (isOverflow: boolean) => void) => {
+  const [isOverflow, setIsOverflow] = useState<boolean>()
+  const ref = useRef<HTMLElement>()
```

Comment:
why we moved the `useRef` call here?

Reply:
Common pattern in hooks that use a reference to an element. Keeps from having to create the reference in the implementation/components